packages/types/src/codec/UInt.ts

```diff
@@ -30,24 +29,26 @@ export default class UInt extends Base<BN> {
 
     this._bitLength = bitLength;
     this._isHexJson = isHexJson;
+    this.raw = this;
   }
 
-  static decodeUInt (value: AnyNumber, bitLength: UIntBitLength): BN {
-    if (value instanceof UInt) {
-      return value.raw;
-    } else if (isHex(value)) {
-      return hexToBn(value);
+  static decodeUInt (value: AnyNumber, bitLength: UIntBitLength): string {
+    // This function returns a string, which will be passed in the BN
+    // constructor. It would be ideal to actually return a BN, but there's a
+    // bug: https://github.com/indutny/bn.js/issues/206.
+    if (isHex(value)) {
+      return hexToBn(value).toString();
     } else if (isU8a(value)) {
       // NOTE When passing u8a in (typically from decoded data), it is always LE
-      return u8aToBn(value.subarray(0, bitLength / 8), true);
+      return u8aToBn(value.subarray(0, bitLength / 8), true).toString();
     } else if (isString(value)) {
-      return new BN(value, 10);
+      return new BN(value, 10).toString();
     }
 
-    return bnToBn(value);
+    return bnToBn(value).toString();
   }
 
-  get bitLength (): UIntBitLength {
+  bitLength (): UIntBitLength {
```

Comment:
Q: Do we actually use/rely on `bitLength` anywhere? (Or is is just there as a "could be useful"). The reason for asking, is that I for the life of me cannot remember why we have it.